src/plugins/discover/public/application/apps/main/components/sidebar/discover_sidebar.tsx

```diff
@@ -224,7 +226,13 @@ export function DiscoverSidebar({
       map.set(parent, value);
     });
     return map;
-  }, [fields, useNewFieldsApi, selectedFields]);
+  };
+
+  const [multiFields, setMultiFields] = useState(() => calculateMultiFields());
+
+  useShallowCompareEffect(() => {
```

Comment:
:information_source: This is the largest speedup in this PR. Previously the `useMemo` triggered on every refresh of Discover (data), since the `fields` and `selectedFields` we got passed in as a dependency where new arrays (though they still contained the same data). Thus causing the memo to recalculate the multi fields maps, thus causing a new array of multi fields being passed into the components further down in this file, thus causing them to rerender, though nothing really changed.

Since the elements within `fields` and `selectedFields` are actually the same reference during re-renders, I use the `useShallowCompareEffect` here, to make sure we're only recalculating multi fields whenever the content of those arrays actually changes.

Reply:
this is great, I see further potential here to omit re-renderings when data doesn't change using `fieldCounts` (not an option today, since it always counts up with every request).  `useCallback` could now be applied to `calculateMultiFields`